src/lightning/fabric/fabric.py

```diff
@@ -653,16 +658,26 @@ def launch(self, function: Optional[Callable[["Fabric"], Any]] = None, *args: An
                 "This script was launched through the CLI, and processes have already been created. Calling "
                 " `.launch()` again is not allowed."
             )
-        if function is not None and not inspect.signature(function).parameters:
+        if function is not _do_nothing:
+            if not callable(function):
+                raise TypeError(
+                    f"`Fabric.launch(...)` needs to be a callable, but got {function}."
+                    " HINT: do `.launch(your_fn)` instead of `.launch(your_fn())`"
+                )
+            if not inspect.signature(function).parameters:
+                raise TypeError(
+                    f"`Fabric.launch(function={function})` needs to take at least one argument. The launcher will"
+                    " pass in the `Fabric` object so you can use it inside the function."
+                )
+        elif isinstance(self.strategy.launcher, (_MultiProcessingLauncher, _XLALauncher)):
             raise TypeError(
-                "The function passed to `Fabric.launch()` needs to take at least one argument. The launcher will pass"
-                " in the `Fabric` object so you can use it inside the function."
+                f"To use the `{type(self.strategy).__name__}` strategy, `.launch()` needs to be called with a function"
```

Comment:
This won't differentiate DDP from DDP spawn unfortunately. Thoughts @awaelchli? Leave it like this or do

```python
if strategy is DDPStrategy:
    strategy_name = "ddp_spawn"
elif strategy is XLAStrategy:
    strategy_name = "xla"
else:
    strategy_name = type(self.strategy).__name__
```

Reply:
I think what you have right now that's fine for now. We could eventually make the registry have a reverse mapping where it gives us the name back so we could construct error messages easier.